projects/core/src/occ/adapters/configurator/variant/converters/occ-configurator-variant-serializer.ts

```diff
@@ -0,0 +1,130 @@
+import { Injectable } from '@angular/core';
+import { Configurator } from '../../../../../model/configurator.model';
+import { Converter } from '../../../../../util/converter.service';
+import { OccConfigurator } from '../occ-configurator.models';
+
+@Injectable()
+export class OccConfiguratorVariantSerializer
+  implements
+    Converter<Configurator.Configuration, OccConfigurator.Configuration> {
+  constructor() {}
+
+  convert(
+    source: Configurator.Configuration,
+    target?: OccConfigurator.Configuration
+  ): OccConfigurator.Configuration {
+    if (target === undefined) {
+      target = { ...(source as any) };
+    }
+    target.complete = source.complete;
+    target.groups = [];
+
+    this.convertGroup(source.attributes, target.groups);
+
+    return target;
+  }
+
+  convertGroup(
+    source: Configurator.Attribute[],
+    occGroups: OccConfigurator.Group[]
+  ) {
+    // Currently only works with product CPQ_LAPTOP
+    // Once groups are supported other products will work as well
+    this.createGroups(occGroups);
+    source.forEach(attribute =>
+      this.mapAttributesToGroups(attribute, occGroups)
+    );
+  }
+
+  convertAttribute(
+    attribute: Configurator.Attribute,
+    occCstics: OccConfigurator.Characteristic[]
+  ): void {
+    const cstics: OccConfigurator.Characteristic = {
```

Comment:
naming: 'cstics' represents only one cstic, so it should be singular 'cstic'

Reply:
Done.